src/newcharacter.cpp

```diff
@@ -1028,31 +1033,38 @@ tab_direction set_traits( const catacurses::window &w, player &u, points_left &p
     // Track how many good / bad POINTS we have; cap both at MAX_TRAIT_POINTS
     int num_good = 0, num_bad = 0;
 
-    std::vector<trait_id> vStartingTraits[2];
+    const int trait_pages = 3; // Number of trait categories: Good, Bad, Neutral, etc.
+    std::vector<trait_id> vStartingTraits[trait_pages];
 
     for( auto &traits_iter : mutation_branch::get_all() ) {
         // Don't list blacklisted traits
         if( mutation_branch::trait_is_blacklisted( traits_iter.first ) ) {
             continue;
         }
 
+        // Always show profession locked traits, regardless of if they are forbidden
+        const std::vector<trait_id> proftraits = g->u.prof->get_locked_traits();
```

Comment:
Don't use `g->u`. This has nothing to do with the global player instance, use the function parameter `u`.